### DISCUSSION TITLE
Error: Cannot find module '@nomicfoundation/hardhat-toolbox'

### DISCUSSION THREAD

**Discussion by TimeKeyRoll** [NomicFoundation/hardhat#2991] [General]
```
Error: Cannot find module '@nomicfoundation/hardhat-toolbox'
```
I am not sure why i am getting this error since i have this file in `hardhat-config-js`.

could anyone please help into it?

![hardhat](https://user-images.githubusercontent.com/101450342/180958707-a74f8a5c-d7bd-4886-9710-6e91bd6aafc1.PNG)

**Comment by fvictorio** [comment]
This means that you haven't installed the Toolbox. Check the instructions [here](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-toolbox#installation).

**Comment by developerasun** [comment]
Check your package.json file and if there is @nomicfoundation/hardhat-toolbox. If not, try to install the module with, 

```sh 
npm install --save-dev @nomicfoundation/hardhat-toolbox
```

and also install peer dependies related to the toolbox module if there is any. Once toolbox installed, your terminal let you know if there is peer dependencies to install.

**Reply by fvictorio** [reply]
Worth mentioning that you only need to install the peer dependencies if you are using an old version of npm or yarn. We recommend using npm 7 or later because it does this automatically and makes creating new hardhat projects way easier.

**Reply by TimeKeyRoll** [reply]
You are right. New way is cool. Tool box seems pretty soothing.

But I'm not sure if toolbox comes with 'solhint' and 'hardhat contract sizer'. Or do I have to download them sperately

**Reply by fvictorio** [reply]
You have to install them separately, yes. We'll be working "soon" (not sure when tbh) on having some boilerplate templates with more stuff, but for project initialization with `npx hardaht` we are trying to strike a balance between something that is fast but complete, which means not including some stuff.

**Reply by TimeKeyRoll** [reply]
Perfecto
